.github/workflows/buildAndTest.yml

```diff
@@ -100,8 +102,9 @@ jobs:
           -DTORCH_MLIR_ENABLE_LTC=OFF \
           -DPython3_EXECUTABLE=$(which python) \
           .
+        cmake --build build
```

Comment:
This is the out-of-tree build step that is now co-located with the cmake configuration for out-of-tree.

Reply:
I would suggest we break the build and test into two and we can choose just the build for all flows

Reply:
Yes, I made these changes this morning (in case you reviewed yesterday). Right now the builds are all colocated in their own step and both unit and integration tests are gated for OOT. Please LMK if I missed anything.

Reply:
I think we should just extract the `cmake --build build` it its own `-name: Build` -- this keeps config step separate from build step. We don't need to specify `--target tools/torch-mlir/all` since that should be default.

Reply:
Ah I see what you mean. Will do.

Reply:
Done. PTAL.